### DISCUSSION TITLE
MDPPerm values

### DISCUSSION THREAD

**Discussion by ofcaah** [MatthiasValvekens/pyHanko#490] [General]
Hi. Currently there's no "0" value in MDPPerm class. While I understand that it's probably due to there being no "0" value in official PDF spec, reality is that '0' seems to be a valid and somewhat useful value. It results in "All changes are allowed" certification, while still providing benefits from certifying signature like change tracking and document's proof of origin. What's your take on this @MatthiasValvekens ?

**Comment by ofcaah** [comment]
after a bit of tweaking with this, some external services seem to complain when DocMDP has been set to 0, however Acrobat Reader still seems to be happy. Perhaps there's another way to get to this outcome? Perhaps not setting the value at all?

**Reply by MatthiasValvekens** [reply]
Hmm, this is not in the PDF standard (and has never been part of any PDF specification AFAIK), so it's going to be tough to convince me to add this.

I struggle to understand why you would even want this, though. Level 3 is already extremely lenient to the point of being unsafe, so I wonder what else you would want to be able to change on top of that. What changes did you have in mind, concretely? Because something tells me that those would invalidate the signature anyway regardless of policy choices.

**Reply by ofcaah** [reply]
terribly sorry for taking so long to reply; I somehow missed the notification of your response.

There's the problem with the same public site that generates signatures not liked by openssl. Currently it decides to break certification when adding a signature (even though there are no apparent changes to the document)

I've decided on ditching certification signature and sticking to plain signature. It yields the same end result: allows to identify the document and protects the document against modifications, while at the same time being way less cumbersome and incompatible.
